internal/repository/targets.go

```diff
@@ -200,6 +200,66 @@ func (r *Repository) UpdateDelegation(ctx context.Context, signer sslibdsse.Sign
 	return state.Commit(r.r, commitMessage, signCommit)
 }
 
+// ReorderDelegations is the interface for the user to reorder rules in gittuf
+// policy.
+func (r *Repository) ReorderDelegations(ctx context.Context, signer sslibdsse.SignerVerifier, targetsRoleName string, ruleNames []string, signCommit bool) error {
+	keyID, err := signer.KeyID()
+	if err != nil {
+		return nil
+	}
+
+	slog.Debug("Loading current policy...")
+	state, err := policy.LoadCurrentState(ctx, r.r, policy.PolicyStagingRef)
+	if err != nil {
+		return err
+	}
+
+	slog.Debug("Loading current rule file...")
+	if !state.HasTargetsRole(targetsRoleName) {
+		return policy.ErrMetadataNotFound
+	}
+
+	targetsMetadata, err := state.GetTargetsMetadata(targetsRoleName)
+	if err != nil {
+		return err
+	}
+
+	slog.Debug("Checking if all rules exist...")
+	for _, ruleName := range ruleNames {
```

Comment:
let's not split checks between here and the policy package, a future refactor could accidentally include one but not the other, etc.